### PR TITLE
Extend the TestCurrencyConverter

### PR DESCRIPTION
**Pull request by chirlu**
Extend the `TestCurrencyConverter` to support EUR↔USD and EUR↔CHF (either direction; not USD↔CHF, though). Also some refactoring to make it simpler.

### CODE REVIEW

**Comment by chirlu** [comment]
I wrote this some time ago because I am going to need it (inverse conversion, and second foreign currency) to test some other change that I hope to finish some other day. I’m submitting it now because #3248 is also trying to add inverse conversion to the `TestCurrencyConverter`, in an ugly way.

File: name.abuchen.portfolio.junit/src/name/abuchen/portfolio/junit/TestCurrencyConverter.java [hunk 1]
```diff
@@ -55,31 +76,24 @@ public String getTermCurrency()
         return termCurrency;
     }
 
-    @Override
-    public Money convert(LocalDate date, Money amount)
```
[the remark this answers is not in the document]
Comment (buchen) [reply name.abuchen.portfolio.junit/src/name/abuchen/portfolio/junit/TestCurrencyConverter.java:59]: It is delegating to the super class, but "just" converts them 1:1 for any currency conversation that are not part of the test currency converter. Has this now become obsolete by introducing CHF explicitly?

[the remark this answers is not in the document]
Comment (chirlu) [reply name.abuchen.portfolio.junit/src/name/abuchen/portfolio/junit/TestCurrencyConverter.java:59]: I don’t understand the remark (or question?). The behaviour (1:1 conversion for unknown currencies) is exactly the same as before:
* Currently (before this change), it is done in this function, `convert()`, in line 69, by creating a new `Money` with the desired currency and the existing amount.
* With this change, `convert()` in the superclass calls our `getRate()` to find the exchange rate, and `getRate()` returns 1 for unknown currency pairs. The superclass `convert()` uses this rate for the conversion, with the same effect.

I don’t actually think any tests rely on this behaviour.

File: name.abuchen.portfolio.junit/src/name/abuchen/portfolio/junit/TestCurrencyConverter.java [hunk 2]
```diff
@@ -55,31 +76,24 @@ public String getTermCurrency()
...
-    {
-        try
-        {
-            return CurrencyConverter.super.convert(date, amount);
-        }
-        catch (MonetaryException e)
-        {
-            // testing: any other currency will be converted 1:1
-            if (!amount.getCurrencyCode().equals(series.getBaseCurrency()))
-                return Money.of(termCurrency, amount.getAmount());
```
Comment (chirlu) [inline name.abuchen.portfolio.junit/src/name/abuchen/portfolio/junit/TestCurrencyConverter.java:69]: Just to point it out, this is where the 1:1 conversion happens in the old code …

File: name.abuchen.portfolio.junit/src/name/abuchen/portfolio/junit/TestCurrencyConverter.java [hunk 3]
```diff
@@ -55,31 +76,24 @@ public String getTermCurrency()
...
+            series = USD_EUR;
+        else if (currencyCode.equals("EUR") && termCurrency.equals("USD"))
+            series = EUR_USD;
+        else if (currencyCode.equals("CHF") && termCurrency.equals("EUR"))
+            series = CHF_EUR;
+        else if (currencyCode.equals("EUR") && termCurrency.equals("CHF"))
+            series = EUR_CHF;
+        else
+            // testing: any other currency will be converted 1:1
+            return new ExchangeRate(date, BigDecimal.ONE);
```
Comment (chirlu) [inline name.abuchen.portfolio.junit/src/name/abuchen/portfolio/junit/TestCurrencyConverter.java:96]: … and this is where, in the new code, an exchange rate of 1 is passed to the superclass’s `convert()`, which uses it to do a 1:1 conversion.

File: name.abuchen.portfolio.junit/src/name/abuchen/portfolio/junit/TestCurrencyConverter.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (buchen) [reply name.abuchen.portfolio.junit/src/name/abuchen/portfolio/junit/TestCurrencyConverter.java:59]: You are right. I forgot (and was looking at the PR only in the browser without full access to the code)
I added the fallback conversation rate later, which makes the overriding of getRate obsolete. Agree.

**Pull request merged**